packages/core/src/Snackbar/StackedSnackbarProvider/StackedSnackbarProvider.js

```diff
@@ -0,0 +1,143 @@
+import React, { forwardRef } from "react";
+import { makeStyles } from "@material-ui/core";
+import PropTypes from "prop-types";
+import { SnackbarProvider, SnackbarContent, useSnackbar } from "notistack";
+
+import HvSnackbarContentWrapper from "../SnackbarContentWrapper";
+
+// We override notistack hook to be able to customize the snackbar that should be called.
+const useHvStackedSnackbar = () => {
+  const { enqueueSnackbar: enqueueNotistackSnackbar, ...others } = useSnackbar();
+  const enqueueSnackbar = (message, options) => {
+    const { id, variant = "success", snackbarContentProps, ...otherOptions } = options;
+    enqueueNotistackSnackbar(
+      JSON.stringify({ message, variant, id, snackbarContentProps }),
+      otherOptions
+    );
+  };
+  return {
+    ...others,
+    enqueueSnackbar,
+  };
+};
```

Comment:
This will return different references on each hook call, that might lead to unwanted re-renders in its users. I suggest:
```suggestion
const useHvStackedSnackbar = () => {
  const { enqueueSnackbar: enqueueNotistackSnackbar, closeSnackbar } = useSnackbar();

  const enqueueSnackbar = useCallback((message, options) => {
    const { id, variant = "success", snackbarContentProps, ...otherOptions } = options;
    enqueueNotistackSnackbar(
      JSON.stringify({ message, variant, id, snackbarContentProps }),
      otherOptions
    );
  }, [enqueueNotistackSnackbar]);

  return useMemo(() => ({
    enqueueSnackbar,
    closeSnackbar
  }), [enqueueSnackbar, closeSnackbar]);
};
```